### ISSUE TITLE
Multiple changes trigger full reset and resync

### ISSUE THREAD

**Issue by viharm** [xbrowsersync/api#46]
### Issue description

Whenever I make many (sometimes even as low as two) changes to my bookmarks, the client triggers a full reset and resync of all bookmarks from the server. Am I doing something wrong? Is there a configuration setting I can change to fix this?

#### API (server)

API service version: `1.1.10` (self hosted docker image).
MongoDB version: `4.2.0` (docker image).
Network: `Bridge`.
OpenMediaVault Docker GUI plugin version: `4.1.6`.
Docker CE server: Engine `19.03.11`; API `1.40`; Go `1.13.10`; Containerd `1.2.13`; runc `1.0.0-rc10`; docker-init `0.18.0`.
Docker CE client: Engine `19.03.11`; API `1.40`; Go `1.13.10`.
Docker Compose version: `1.8.0-2`.
OS: `OpenMediaVault 4.1.35-1` (`Arrakis`; based on `Debian GNU/Linux 9.12 Stretch`).
Linux kernel: `4.19.0-0.bpo.8-amd64 #1 SMP Debian 4.19.98-1~bpo9+1 (2020-03-09) x86_64 GNU/Linux`.
Filesystem: `ext4`.

#### Client

Client add-on version: `1.5.2`.
Browser: `Firefox` x86-64, `78.0.1`.
Operating System: `Kubuntu 20.04`.
KDE Plasma Version: `5.18.5`.
KDE Frameworks Version: `5.68.0`.
Qt Version: `5.12.8`.
Kernel Version: `5.4.0-40-generic`.
OS Type: `64-bit`.
Processors: 4× `Intel® Celeron® CPU N3150 @ 1.60GHz`.

**Comment by nero120** [comment]
This is likely an extension issue not API, can you post the log from the extension please?

**Comment by viharm** [comment]
> 
> 
> This is likely an extension issue not API, can you post the log from the extension please?

I'm afraid I don't know how to get the logs from extensions. Can you please point me in the right direction?

**Comment by nero120** [comment]
> I'm afraid I don't know how to get the logs from extensions. Can you please point me in the right direction?

Open the xBrowserSync extension window, click on the cog icon in the top right to get to the Settings panel, then scroll down to Issues and click the Download Log button.

**Comment by protocolist** [comment]
I have this same issue and i'm happy to assist with troubleshooting if you need.

**Comment by nero120** [comment]
@protocolist can you post your log please?

**Comment by protocolist** [comment]
Sure. I cleared the log and then attempted an operation where I moved three bookmarks from one folder to another. This immediately triggered the reset and sync, and the three bookmarks remained in the original location. Browser is Edge 84.0.522.40, I have also experienced the problem on ungoogled-chromium 81.0.4044.138.

Let me know if you need more detail.

[xbs_log_20200731002309.txt](https://github.com/xbrowsersync/api/files/5001808/xbs_log_20200731002309.txt)

**Comment by nero120** [comment]
@protocolist yes the log confirms it was a multiple move that triggered the re-sync. did you move the bookmarks to a folder that was a sibling of the bookmarks?

**Comment by protocolist** [comment]
That's correct. I've just done some experimentation moving the same three bookmarks around:

Moved down to a nested folder from its parent folder - triggers reset
Moved up from a nested folder to its parent folder - triggers reset
Moved up from a nested folder to a different folder at the same hierarchical level as the parent folder - okay
Moved up from a nested folder to the root folder - okay

I then reduced the number of bookmarks I was attempting to move to two, same as above

Finally, I reduced the number to one, same problem.

As such, I think that it's possible i'm reporting a different bug to the original poster, or alternatively, this issue is typically only noticed while moving multiple bookmarks.

Log from a single move is attached. I did a succesful move of the same object just before in this same log.
[xbs_log_20200731082101.txt](https://github.com/xbrowsersync/api/files/5004054/xbs_log_20200731082101.txt)

**Comment by viharm** [comment]
@protocolist , unfortunately I haven't been able to test it systematically as you have. However the symptoms are very similar. Are you using your own API server, or an existing public service?

@nero120 , unfortunately, I had to upgrade my OMV installation, so couldn't get to my log as requested. I will do so as soon as I can. However I can confirm that the symptoms are very similar to those described by @protocolist .

**Comment by protocolist** [comment]
I'm on the official API, https://api.xbrowsersync.org/

**Comment by nero120** [comment]
@viharm, this issue is not linked to the API, it's a bug in the app. This issue is in the wrong repo so I will move it to app. Can you please do the following test to confirm:

1. Whilst synced, open the native browser bookmark manager.
2. In any folder, create two bookmarks and one folder.
3. Select both bookmarks and drag them into the sibling folder.

@protocolist your issue is actually already logged here: #158. I've already fixed it in dev and will be included in the upcoming release (v1.6.0).
